.github/workflows/ci.yml

```diff
@@ -70,11 +70,14 @@ jobs:
       - name: Init cache
         uses: Swatinem/rust-cache@v2
 
-      - name: Install Fuel toolchain
-        uses: FuelLabs/action-fuel-toolchain@v0.6.0
+      - name: Install fuel toolchain
+        uses: cachix/install-nix-action@v20
         with:
-          toolchain: latest
-          date: 2023-04-12
+          nix_path: nixpkgs=channel:nixos-unstable
+          extra_nix_config: |
+            extra-substituters = https://fuellabs.cachix.org
+            extra-trusted-public-keys = fuellabs.cachix.org-1:3gOmll82VDbT7EggylzOVJ6dr0jgPVU/KMN6+Kf8qx8=
```

Comment:
```suggestion
          extra_nix_config: |
            substituters = https://fuellabs.cachix.org
            trusted-public-keys = fuellabs.cachix.org-1:3gOmll82VDbT7EggylzOVJ6dr0jgPVU/KMN6+Kf8qx8=
```

Reply:
Hmm from [this](https://nixos.org/manual/nix/stable/command-ref/conf-file.html) docs, it looks like providing `extra_` prefix to nix config tells nix to append the given public key and substituters, rather than overriding already existing ones. In this case we overriden nixos cache and made the process slower i guess. (Since we had packages found from nixos cache rather than ours but with this change we only find the packages in our cache and re-build other ones). The default substituters is `https://cache.nixos.org`

Reply:
do we need to enable experimental features here?

Reply:
I think the action has the flake experimental feature enabled by default (last bullet point in the linked section):

https://github.com/cachix/install-nix-action#features

Reply:
Yeah I wasn't sure if that was being overridden due to this logic: 
![image](https://github.com/kayagokalp/sway-applications-nix-ci/assets/794823/e00281d5-492d-434f-a563-7c38ec0b7b30)

https://github.com/cachix/install-nix-action/blob/16b951426ed4927ca0d26d32e8f5a10638579ec8/install-nix.sh#L31